include/unified_teleop/pnt_stmp_modifiers.hpp

```diff
@@ -0,0 +1,217 @@
+        // This check allows the function to simply normalize the vector without scaling it if new_mag is 0
+        if (new_mag == 0)
+        {
+            new_mag = magnitude;
+        }
+
+        // Check to ensure the current magnitude is not 0 to avoid division by zero
+        if (magnitude != 0)
+        {
+            // Adjust each component of the vector (x, y, z) to the new magnitude
+            // The formula used is: (component / current magnitude) * new magnitude
+            // This scales the vector to have a magnitude of new_mag while maintaining its direction
+            normalized_pnt_stmp.point.x = new_mag * normalized_pnt_stmp.point.x / magnitude;
+            normalized_pnt_stmp.point.y = new_mag * normalized_pnt_stmp.point.y / magnitude;
+            normalized_pnt_stmp.point.z = new_mag * normalized_pnt_stmp.point.z / magnitude; 
+        }
+
+        // Return the normalized (and possibly scaled) PointStamped message
+        return normalized_pnt_stmp;
+    }
+
+    /// @brief Rounds the coordinates of a PointStamped message to a specified number of decimal places.
+    /// @param input_command The PointStamped message whose coordinates will be rounded.
+    /// @param num_decimal_places The number of decimal places to round to.
+    /// @return The PointStamped message with rounded coordinates.
+    geometry_msgs::msg::PointStamped round_pntstmp(geometry_msgs::msg::PointStamped input_command, int num_decimal_places)
+    {
+        geometry_msgs::msg::PointStamped new_command;
+        new_command.point.x = round(input_command.point.x * std::pow(10, num_decimal_places))/std::pow(10, num_decimal_places);
+        new_command.point.y = round(input_command.point.y * std::pow(10, num_decimal_places))/std::pow(10, num_decimal_places);
+        new_command.point.z = round(input_command.point.z * std::pow(10, num_decimal_places))/std::pow(10, num_decimal_places);
+        return new_command;
+    }
+
+    /// @brief The specific axis in a PointStamped message that will be modified
+    enum class AxisType
+    {
+        X_Axis,
+        Y_Axis,
+        Z_Axis
+    };
+
+    /// @brief Adjusts the value of a specific axis of a PointStamped message based on controller input, increasing or decreasing the value as specified.
+    /// @param input The controller input that determines how the original message will be modified.
+    /// @param orig_message The original PointStamped message to be modified.
+    /// @param axis_type The axis (X, Y, Z) of the PointStamped message to be adjusted.
+    /// @param is_increasing Boolean indicating whether to increase (true) or decrease (false) the axis value.
+    /// @param joy_state The current state of the controller inputs.
+    /// @return A modified PointStamped message with adjusted values for the specified axis.
+    geometry_msgs::msg::PointStamped adjust_mirror_joy(const std::optional<rosnu::MovementInput> input,
```

Comment:
This function should not rely on ROS running to work. So no shutdown, ros logging etc.

Reply:
I'm assuming you're RCLCPP_ERROR also falls within that category? I'm removing it (Can't find anything else you could be referring to)